airbyte-integrations/connectors/destination-langchain/destination_langchain/indexer.py

```diff
@@ -69,16 +69,33 @@ def __init__(self, config: PineconeIndexingModel, embedder: Embedder):
         self.embed_fn = measure_time(self.embedder.langchain_embeddings.embed_documents)
 
     def pre_sync(self, catalog: ConfiguredAirbyteCatalog):
+        index_description = pinecone.describe_index(self.config.index)
+        self._pod_type = index_description.pod_type
         for stream in catalog.streams:
             if stream.destination_sync_mode == DestinationSyncMode.overwrite:
-                self.pinecone_index.delete(filter={METADATA_STREAM_FIELD: stream.stream.name})
+                self._delete_vectors({METADATA_STREAM_FIELD: stream.stream.name})
 
     def post_sync(self):
         return [AirbyteMessage(type=Type.LOG, log=AirbyteLogMessage(level=Level.WARN, message=self.embed_fn._get_stats()))]
 
+    def _delete_vectors(self, filter):
+        if self._pod_type == "starter":
+            # Starter pod types have a maximum of 1000000 rows
+            top_k = 10000
```

Comment:
why 10 000 if there can be 100 000 rows?

Reply:
Because for querying, the limit is 10k... I know this isn't great, but for evaluation purposes (which is the point of the starter pod), it should be enough (discussed this with @rschwabco from Pinecone) and adjusted the documentation.